src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,378 @@
+                )
+
+            # Go through past timeslices
+            for n in range(0, leadtime):
+                # First step back, so we use i,j coords to find out parcel location
+                # in terms of array point
+                if n == 0:
+                    x = lon_pnt
+                    y = lat_pnt
+                    z = plev_idx
+
+                # Only seek preceding wind if its inside domain..
+                if not outside_lam:
+                    # Get vector profile at current time - nearest whole gridpoint.
+                    u = x_arr[leadtime - n, int(z), int(y), int(x)]
+                    v = y_arr[leadtime - n, int(z), int(y), int(x)]
+                    w = z_arr[leadtime - n, int(z), int(y), int(x)]
+                    g = g_arr[leadtime - n, int(z), int(y), int(x)]
+
+                    # First, compute horizontal displacement using inverse of horizontal vector
+                    # Convert m/s to m/[samplingrate]h, then m ->  model gridpoints
+                    if timeunit == "hour":
+                        du = ((u * 60 * 60 * dt) / ew_spacing) * -1.0
+                        dv = ((v * 60 * 60 * dt) / ns_spacing) * -1.0
+                        dz = (w * 60 * 60 * dt) * -1.0
+
+                    # Get column of geopot height.
+                    g_col = g_arr[(leadtime - n), :, int(y), int(x)]
+
+                    # New geopotential height of parcel - store 'capacity' between timesteps as vertical motions smaller.
+                    if n == 0:
+                        new_g = g + dz
+                        pre_g = new_g
+                    else:
+                        new_g = pre_g + dz
+
+                    # Calculate which geopot level is closest to new geopot level.
+                    z = np.argmin(np.abs(g_col - new_g))
+
+                    # Update x,y location based on displacement. Z already updated
+                    x = x + du
+                    y = y + dv
+
+                    # If it is now outside domain, then save age and dont process further with outside lam flag
+                    # Support cyclic domains like K-SCALE, where x coord out of domain gets moved through dateline.
+                    if cyclic:
+                        if (
+                            x <= -1
+                        ):  # as for example -0.3 would still be in domain, but x_arr.shape-0.3 would result in index error
+                            x = x_arr.shape[3] + x  # wrap back around dateline
+                        elif x >= x_arr.shape[3]:
+                            x = x_arr.shape[3] - x
+                    else:
+                        if x < 0 or x >= x_arr.shape[3]:
+                            ageofair_local[leadtime, lat_pnt] = n * dt
+                            outside_lam = True
+
+                    if y < 0 or y >= x_arr.shape[2]:
+                        ageofair_local[leadtime, lat_pnt] = n * dt
+                        outside_lam = True
```

Comment:
This should probably be its own function.

Reply:
I think if this section became a separate function, the aoa_core would just compute spacings and iterate over points. If we separate this out, it might be harder to visually follow/debug, given I'd have to pass quite a few variables to a new function? I.e. balance between functions doing a particular task, and readability

Reply:
Fair enough.